### DISCUSSION TITLE
Warnings in ly1.x not in ly2.x (fatal hash)

### DISCUSSION THREAD

**Discussion by flyflick** [CESNET/libyang#1867] [Q&A]
With our usage of libyang 1.0.240 we would see the following error:

2022-05-11T17:41:38.340+0000 - INFO - netconf - netopeer2 - [polt_entrypoint.py:210] - 2022-05-11 17:41:38:[WRN]: LY: Augmentations from modules "bbf-pppoe-intermediate-agent" and "bbf-xpon" have fatal hash collision.
2022-05-11T17:41:38.340+0000 - INFO - netconf - netopeer2 - [polt_entrypoint.py:210] - 2022-05-11 17:41:38:[WRN]: LY: It will cause no errors if module "bbf-pppoe-intermediate-agent" is always loaded before "bbf-xpon".

With the same code and usage of libyang 2.0.164, we no longer see this warning.

Also noticed that new 2.x libyang no longer has the function: lyb_check_augment_collision()

Is this becasue the functionality has changed such that this is no longer a possible issue?

Thanks for your answer in advance!

**Comment by michalvasko** [comment]
Mostly, we were just being extra careful in *libyang* v1 and printed this message while I am not sure the scenario can even occur (that you would load the augment is a different order). So, strictly speaking the situation is the same but this check was simply removed.

**Reply by flyflick** [reply]
OK, so the augment clash still occurs in libyang 2.x, but no warning is printed...
And if I interpret correctly, you are saying even changing the order would not fix the fatal hash?
Or maybe you are saying it is not a "fatal hash" problem to begin with?

**Reply by michalvasko** [reply]
A problem could only occur if the situation described would happen, `if module "bbf-pppoe-intermediate-agent" is not loaded before "bbf-xpon"`. But the order in which the *sysrepo* modules are being loaded and compiled does not change.

**Reply by flyflick** [reply]
Hi Michal,

When you say loading, are we talking about:
a) sysrepoctl --install
-or-
b) nc_ly_ctx_load_module()

And I'm not sure what is meant by "order in which the sysrepo modules are being loaded and compiled does not change".

**Reply by michalvasko** [reply]
I am talking about `nc_ly_ctx_load_module()` order, which will not change once the modules are installed.
